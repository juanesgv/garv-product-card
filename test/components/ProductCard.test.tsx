import * as React from 'react';
import rendered from 'react-test-renderer'
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductCard', ()=>{

    test('Debe mostrar el componente correctamente', ()=>{
        const wrapper = rendered.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <h1>Product card</h1>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
    
    test('Debe fucnionar el contador', ()=>{
        const wrapper = rendered.create(
            <ProductCard product={product2}>
                {
                    ({count, increaseBy}) => (
                        <>
                            <h1>Productcard</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(1)}></button>
                        </>
                    )
                }
            </ProductCard>
        )

        let tree = wrapper.toJSON()
        expect(tree).toMatchSnapshot()

        // (tree as any).children[2].pops.onClick()

        // tree = wrapper.toJSON()

        // expect((tree as any).children[1].children[0].toBe('1'))
    })
    
})
