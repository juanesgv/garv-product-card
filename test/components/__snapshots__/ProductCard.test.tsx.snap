// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe fucnionar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Productcard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard Debe mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;
